src/rpc/blockchain.cpp

```diff
@@ -2140,7 +2142,11 @@ UniValue scantxoutset(const JSONRPCRequest& request)
                 if (!desc->Expand(i, provider, scripts, provider)) {
                     throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, strprintf("Cannot derive script without private keys: '%s'", desc_str));
                 }
-                needles.insert(scripts.begin(), scripts.end());
+                for (const auto script : scripts) {
+                    std::string desc = InferDescriptor(script, provider)->ToString();
```

Comment:
Nit: Use another variable name to avoid showing `desc` declared on line 2133 (`auto desc = Parse(desc_str, provider);`)?